### PR TITLE
Add setting up a dev environment notes

### PR DESCRIPTION
**Pull request by aviramha**
Following discussions on https://github.com/PyO3/pyo3/issues/2084

### CODE REVIEW

File: Contributing.md [hunk 1]
```diff
@@ -15,6 +15,16 @@ You can browse the API of the non-public parts of PyO3 [here](https://pyo3.rs/in
 
 The following sections also contain specific ideas on where to start contributing to PyO3.
 
+## Setting up a development environment
+
+To work and develop PyO3, you need Python & Rust installed on your system.
+* We encourage the use of [rustup](https://rustup.rs/) to be able to select and choose specific toolchains based on the project.
+* [Pyenv](https://github.com/pyenv/pyenv) is also highly recommended for being able to choose a specific Python version.
```
Comment (davidhewitt) [inline Contributing.md:22]: Actually on Ubuntu I've just used the `deadsnakes` repository to install Python versions 3.7 through 3.10 and then have made virtualenvs for each of them. Perhaps that can be added as an alternative option?

(Having multiple virtualenvs also works on Windows, where `pyenv` does not work at all.)

**Review by davidhewitt — COMMENTED** [review]
Thanks for kicking these notes off! Please can a little extra detail be added (see below)...

File: Contributing.md (hunk 1 above)
Comment (aviramha) [reply Contributing.md:22]: lmk what you think now

**Review by davidhewitt — APPROVED** [review]
Great, thanks!

**Pull request merged**